blosc/blosc2.h

```diff
@@ -20,9 +20,29 @@
 #include <stdlib.h>
 #include <stdint.h>
 #include <stdbool.h>
+#include <stdio.h>
 #include "blosc2-export.h"
 #include "blosc2-common.h"
 
+#if defined(_WIN32) && !defined(__MINGW32__)
+#include <windows.h>
+  #include <malloc.h>
+
+/* stdint.h only available in VS2010 (VC++ 16.0) and newer */
+  #if defined(_MSC_VER) && _MSC_VER < 1600
+    #include "win32/stdint-windows.h"
+  #else
+    #include <stdint.h>
+  #endif
+
+  #define fseek _fseeki64
```

Comment:
Based on the definition of [_fseeki64](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/fseek-fseeki64?view=msvc-160) using `__int64` your definition of `blosc2_seek_cb` might have a problem.

Reply:
In this case, It seems this is the problem :)

Otherwise, this definition is also used in the `master` branch:

https://github.com/Blosc/c-blosc2/blob/f489cdbd0a2567e07365671618fd859a31eec4ae/blosc/blosc2.c#L71

Maybe we have found why are some random failures on Win32.

@FrancescAlted, do you know the reasons for introducing this definition?

Reply:
IIRC I have introduced `_fseeki64` so as to support files larger than 2 GB on Win32.  But obviously this has some undesirable consequences (see also https://stackoverflow.com/questions/4003405/32-bit-windows-and-the-2gb-file-size-limit-c-with-fseek-and-ftell).  In addition, Microsoft is not providing Win32 for OEMS for a year now (https://www.engadget.com/windows-10-32-bit-oem-173055990.html).  Hence, if there is no opposition, I very much prefer to drop the support of Win32 entirely.

Reply:
@FrancescAlted I'm not sure if `_fseeki64` has anything to do with Win32 or Win64. I always believed that `fseek` on Windows only allowed 2GB offset and `_fseeki64` allowed greater than 2GB offset regardless of the subsystem being used. The size of long (for offset arg) also does not change when compiling on 32-bit Windows or 64-bit Windows afaik. The original code may be correct, but we may need to find another way to do it.. like change the definition of `blosc2_seek_cb` when `fseek64i` is used.

Reply:
Ok, thanks. Maybe @aleixalcacer would want to have another attempt at using int64_t, but if that doesn't work, I would go ahead and drop win32 support.